Replace `<stuff>' with '<stuff>'

owen-mc-diffblue opened the pull request:
This changes the error message from
failed to load class `java.nio.file.Path'
to
failed to load class 'java.nio.file.Path'

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

Commit:
Do not use backtick as opening single quote

Replace it with '. For example, This changes an error message from
failed to load class `java.nio.file.Path'
to
failed to load class 'java.nio.file.Path'

The reason is that the convention of using ` for opening single
quotes looks odd in most modern fonts. For more discussion please
read:

https://www.cl.cam.ac.uk/~mgk25/ucs/quotes.html

danpoe commented:
That's probably intentional as the pattern with left hand ``` ` ``` and right hand `'` occurs more frequently in the code than two single quotes.

owen-mc-diffblue commented:
Thanks for pointing that out @danpoe , on closer inspection I see that you are right. In that case, I'm expanding the scope of this PR to replace ` with ' everywhere where the closing delimiter is '. I think this is a much more common usage of punctuation.

tautschnig commented:
@owen-jones-diffblue Why is this a desirable change?

kroening commented:
This is certainly a very unimportant point one could get very excited about.
https://www.cl.cam.ac.uk/~mgk25/ucs/quotes.html
We could add `quotebegin` and `quoteend` stream modifiers. The result could then be quotes, a change in font, etc., depending on target.

owen-mc-diffblue commented:
@tautschnig As the link that @kroening posted says, the convention of quoting `text like this' looks odd in most modern fonts.

I appreciate this is not a majorly important change, but it's something that annoys me. I think it makes the output look unprofessional. It was a treat I gave myself after doing some very tedious work I didn't really want to do. The prompt was someone asking about an error message, and they obviously thought it was a mistake that there was a ` instead of a '.

tautschnig commented:
@owen-jones-diffblue Thank you for elaborating - it would be really helpful to have the "Why?" answered in commit messages (always, not just here). That settled, we are probably very inconsistent about our quoting and thus this is a really good step forward! My only reservation is that there may be users parsing warnings and thus we'd (negatively) surprise them, but arguably no one should rely on formatting of plain-text messages so I'm ok to ignore this one.

Commit:
Remove erroneous backtick

tautschnig commented on jbmc/regression/jbmc/generics_type_param/test.desc:
```diff
@@ -7,6 +7,6 @@ Reading class AWrapper
-failed to load class \`IWrapper\'
+failed to load class \'AWrapper\'
+failed to load class \'FWrapper\'
+failed to load class \'IWrapper\'
```
Nit picking: why are we using escapes here?

owen-mc-diffblue replied:
I agree, and it's probably closely enough related to this PR to include it

peterschrammel reviewed (commented):
I actually like @kroening's suggestion of stream modifiers... For the next rainy day...

Commit:
Do not quote classes we fail to load

The error message used to be
  failed to load class 'Test'
This commit makes it
  failed to load class Test
The reasoning is that there is no need to quote the class name, it is
totally clear where it starts and where it ends.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 8e86fe201a0277ffcff2837248b2fe0b68f09838).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/118429422

chrisr-diffblue reviewed (approved):
Looks good to me. I'd also +1 on the idea of having stream modifiers. That would be very helpful for the use-case of wanting to be able to 'redact' the output of tools, such as in cases where we would like to have the end-user be able to send us output from the tools, but where they do not want to leak any information about their own input source code. With stream modifiers it would be simple to add a redaction mode or other tokens that could be used for post-processing redaction.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 4c900d1bf8dda708f9b63812114faecf366490f5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/121070770

owen-mc-diffblue commented:
TG bump passed CI
